### ISSUE TITLE
Extend AppD application definition to support describing app channels and their purpose

### ISSUE THREAD

**Issue by kriswest** [finos/FDC3#371]
## Enhancement Request
Add fields to describe channels with predetermined names and usage to the description of an App that creates and/or uses them.

### Use Case:
Some applications with FDC3 integration choose to communicate on predetermined app channel names, rather than on user-controlled system channels. This is more common where the integrated application is a singleton, e.g. a view from an Order Management System (OMS) which broadcasts details of the currently selected order, for other apps to use as context. There is currently no way to advertise the channel names used (either via the desktop agent or AppD). Hence, discovery of these channel names currently requires reference to the application's own documentation, which will be different for every application and vendor.

By providing a standardized location to advertise predetermined channel names in the AppD we can encourage firms hosting AppDs to make discovery easier for developers and/or users (where user's are given tools to work with named channels directly) by using the suggested field.   

```JSON
{
  "appId": "string",
  "name": "string",
  ...
  "channels": [
    {
      "name": "myChannel",
      "description": "Description of what this channel is used for"
    },
    {
      "name": "myAppsOrders",
      "description": "Each time a user selects an order in the user interface a summary of the order information is broadcast on this channel as an fdc3.order Object."
    }
  ],
  ...
}
```

**Comment by rikoe** [comment]
I assume this is opt-in @kriswest? There are some use cases where apps may want to keep the channel private...

**Comment by kriswest** [comment]
Yes, it should definitely be optional - I just intended it as a prod to share the information in a consistent location.

**Comment by thorsent** [comment]
We have a proposal on the way that describes a more generalized approach to advertising  of channels that provide utility services. While this issue isn't in conflict with that approach, I feel that discussion around the broader concept is prudent because it may impact the ideal for AppD schema.

We should also be careful to consider how advertisement of channel purpose coincides with advertisement of context listening/broadcast.

**Comment by kriswest** [comment]
Merged into issue #247 which has a similar use-case and now supercedes this issue
